### PR TITLE
Use node logging in moveit_ros

### PR DESCRIPTION
**Pull request by tylerjw**
### Description

I updated the logger util library with some changes

I removed the assigning into the return of a function. It looks weird and I replaced it with a `set_logger` function.

I added more static state so if `get_logger` is called before `set_logger` it will still work as it will create a `moveit` node. There might be a better way to do this. I did this so child loggers created without creating the base node would work. I added a test for the child logger without first assigning the node logger.

### CODE REVIEW

**Comment by tylerjw** [comment]
#2376

File: moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp [hunk 1]
```diff
@@ -75,23 +76,23 @@ RobotModelLoader::~RobotModelLoader()
 
 namespace
 {
-bool canSpecifyPosition(const moveit::core::JointModel* jmodel, const unsigned int index)
+bool canSpecifyPosition(const moveit::core::JointModel* jmodel, const unsigned int index, const rclcpp::Logger& logger)
```
Comment (henningkayser) [inline moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp:79]: ouch

**Review by henningkayser — APPROVED** [review]
Very clean refactoring. The changes are much less intrusive than I thought. Please update https://github.com/ros-planning/moveit2/blob/main/doc/MIGRATION_GUIDE.md before merging.

File: moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp (hunk 1 above)
Comment (tylerjw) [reply moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp:79]: this is one of the trade-offs of carrying around state everywhere for logging. for the child loggers you have to pass them around (either by putting them in shared state in a class or by using function parameters).

Comment (tylerjw) [reply moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp:79]: For free functions that are public I think we should just use the global node logger and users can use `{function}` to filter for those logs.

For functions like this one that are in the private namespace of cpp file, adding parameters to it has no affect on the public api and seems fine to me.

**Pull request merged**